modules/core/src/main/java/org/locationtech/jts/geom/GeometryFactory.java

```diff
@@ -602,11 +602,17 @@ public LineString createLineString(CoordinateSequence coordinates) {
   public Geometry createGeometry(Geometry g)
   {
     GeometryEditor editor = new GeometryEditor(this);
-    return editor.edit(g, new GeometryEditor.CoordinateSequenceOperation() {
-      public CoordinateSequence edit(CoordinateSequence coordSeq, Geometry geometry) {
-                  return coordinateSequenceFactory.create(coordSeq);
-          }
-    });
+    return editor.edit(g, new CloneOp(coordinateSequenceFactory));
+  }
+
+  private static class CloneOp extends GeometryEditor.CoordinateSequenceOperation {
```

Comment:
Can this be called CoordSeqCloneOp, just to leave some future naming freedom?

Reply:
Sure, fixed.